### PR TITLE
implement simple TUI

### PR DESCRIPTION
**Pull request by jacobhenn**
it works! it turns out it was less of a bug in the program and more of a bug in the way i was testing it. so currently the TUI lets you crudely monitor raw incoming messages from the daemon and send messages to the daemon. it requires that the daemon, or some imitation of it, is connected to both fifos before it can do anything. if you don't want to actually use the daemon for this, you can just `cat` from `ctodbuf` in a while loop and `echo` bytes of your choice into `dtocbuf` (even if the messages are invalid it lets the client know there's someone on the other end).

### CODE REVIEW

File: client/src/main.rs [hunk 1]
```diff
@@ -1,32 +1,229 @@
...
-            Err(e) => println!("recieved invalid message: {e}"),
+fn chat() -> Result<()> {
+    let mut state = State::new()?;
+    state.start()?;
+
+    Ok(())
+}
+
+#[allow(unused_must_use)]
+/// Try our best to clean up the terminal state; if too many errors happen, just print some newlines and call it good.
```
Comment (adamhutchings) [inline client/src/main.rs:204]: Average frontend developer

Comment (jacobhenn) [reply client/src/main.rs:204]: i copied that code almost exactly from `guac`

**Review by JakeRoggenbuck — APPROVED** [review]
Very nice!

File: client/src/main.rs [hunk 2]
```diff
@@ -1,32 +1,249 @@
...
+    /// a handle to the `~/.rclc/dtocbuf` fifo in read-only mode
+    dtocbuf: File,
+
+    /// a handle to the `~/.rclc/ctodbuf` fifo in write-only mode
+    ctodbuf: File,
+}
+
+impl<'a> State<'a> {
+    fn new() -> Result<Self> {
+        let mut stdout = io::stdout().lock();
```
Comment (Sophon96) [inline client/src/main.rs:39]: This doesn't compile for me (borrow checker errors).

**Pull request merged**